### PR TITLE
feat(j-s): Send subpoena intro text to service portal

### PR DESCRIPTION
**Pull request by oddsson**
# Send subpoena intro text to service portal

[Asana](https://app.asana.com/0/1199153462262248/1208558070600286/f)

## What

Start sending text to the service portal that explains what kind of a subpoena the accused is receiving and what happens if they don't show up for the hearing. 

## Why

This text needs to be gendered and therefore it's easier for us to send the text, rather than the service portal having to choose the correct text based on the accused gender.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Summary by CodeRabbit

- **New Features**
	- Streamlined handling of introductory messages for subpoenas based on gender.
	- Enhanced API response structure for subpoenas, including new properties for detailed information.

- **Bug Fixes**
	- Improved error handling for defender assignments in the update process.

- **Documentation**
	- Added localized message definitions for various court-related scenarios, enhancing internationalization.

- **Chores**
	- Updated module exports to include new message definitions for better accessibility.

### CODE REVIEW

File: apps/judicial-system/digital-mailbox-api/src/app/modules/cases/case.service.ts [hunk 1]
```diff
@@ -103,10 +123,13 @@ export class CaseService {
     defendantNationalId: string,
     lang?: string,
   ): Promise<SubpoenaResponse> {
+    await this.refreshFormatMessage()
     const caseData = await this.fetchCase(caseId, defendantNationalId)
+
     return SubpoenaResponse.fromInternalCaseResponse(
       caseData,
       defendantNationalId,
+      this.formatMessage,
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/digital-mailbox-api/src/app/modules/cases/case.service.ts:132]: No thanks.

File: apps/judicial-system/digital-mailbox-api/src/app/modules/cases/case.service.ts [hunk 2]
```diff
@@ -30,8 +33,25 @@ export class CaseService {
...
+  private async refreshFormatMessage(): Promise<void> {
+    return this.intlService
+      .useIntl(['judicial.system.backend'], 'is')
+      .then((res) => {
+        this.formatMessage = res.formatMessage
+      })
+      .catch((reason) => {
+        this.logger.error('Unable to refresh format messages', { reason })
+      })
+  }
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/digital-mailbox-api/src/app/modules/cases/case.service.ts:53]: No thanks

**Review by gudjong — CHANGES_REQUESTED** [review]
Looks good, except for the Contentful strings that were moved to a lib.

File: libs/judicial-system/consts/src/lib/messages.ts [hunk 3]
```diff
@@ -0,0 +1,138 @@
+import { defineMessage, MessageDescriptor } from '@formatjs/intl'
```
[the remark this answers is not in the document]
Comment (oddsson) [reply libs/judicial-system/consts/src/lib/messages.ts]: 🙏

**Review by gudjong — APPROVED** [review]

**Pull request merged**